packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/style/RadialGradient.kt

```diff
@@ -0,0 +1,377 @@
+    )
+
+    if (positionMap == null) {
+      return@run defaultPosition
+    }
+
+    var top: LengthPercentage? = null
+    var left: LengthPercentage? = null
+    var right: LengthPercentage? = null
+    var bottom: LengthPercentage? = null
+
+    if (positionMap.hasKey("top")) {
+      val rawTop = positionMap.getDynamic("top")
+      top = LengthPercentage.setFromDynamic(rawTop)
+    } else if (positionMap.hasKey("bottom")) {
+      val rawBottom = positionMap.getDynamic("bottom")
+      bottom = LengthPercentage.setFromDynamic(rawBottom)
+    } else {
+      return@run defaultPosition
+    }
+
+    if (positionMap.hasKey("left")) {
+      val rawLeft = positionMap.getDynamic("left")
+      left = LengthPercentage.setFromDynamic(rawLeft)
+    } else if (positionMap.hasKey("right")) {
+      val rawRight = positionMap.getDynamic("right")
+      right = LengthPercentage.setFromDynamic(rawRight)
+    } else {
+      return@run defaultPosition
+    }
+
+    Position(top, left, right, bottom)
+  }
+
+  private val size: GradientSize = run {
+    if (sizeMap == null) {
+      return@run GradientSize(
+        GradientSize.SizeType.KEYWORD,
+        SizeKeyword.FARTHEST_CORNER
+      )
+    }
+
+    if (sizeMap.hasKey("keyword")) {
+      val keywordString = sizeMap.getString("keyword")
+      val keyword = SizeKeyword.fromString(keywordString)
+      return@run GradientSize(
+        GradientSize.SizeType.KEYWORD,
+        keyword
+      )
+    }
+
+    if (sizeMap.hasKey("x") && sizeMap.hasKey("y")) {
+      val xDynamic = sizeMap.getDynamic("x")
+      val yDynamic = sizeMap.getDynamic("y")
+      val x = LengthPercentage.setFromDynamic(xDynamic)
+      val y = LengthPercentage.setFromDynamic(yDynamic)
+
+      if (x != null && y != null) {
+        return@run GradientSize(
+          GradientSize.SizeType.DIMENSIONS,
+          GradientSize.Dimensions(x, y)
+        )
+      }
+    }
+
+    GradientSize(
+      GradientSize.SizeType.KEYWORD,
+      SizeKeyword.FARTHEST_CORNER
+    )
+  }
+
+  private val colorStops: ArrayList<ColorStop> = run {
+    val stops = ArrayList<ColorStop>(colorStopsArray.size())
+    for (i in 0 until colorStopsArray.size()) {
+      val colorStop = colorStopsArray.getMap(i) ?: continue
+      val color: Int? =
+        when {
+          !colorStop.hasKey("color") || colorStop.isNull("color") -> {
+            null
+          }
+
+          colorStop.getType("color") == ReadableType.Map -> {
+            ColorPropConverter.getColor(colorStop.getMap("color"), context)
+          }
+
+          else -> colorStop.getInt("color")
+        }
+
+      val position = LengthPercentage.setFromDynamic(colorStop.getDynamic("position"))
+      stops.add(ColorStop(color, position))
+    }
+    stops
+  }
+
+  fun getShader(width: Float, height: Float): Shader {
+    var centerX: Float = width / 2f
+    var centerY: Float = height / 2f
+    
+    if (position.top != null) {
+      centerY = position.top.resolveToPixel(height)
```

Comment:
For this and all other usages of resolveToPixel

```suggestion
      centerY = position.top.resolve(height).dpToPx()
```